### PR TITLE
chore: Add GitHub workflow for detecting changes to `aws-amplify` package.json

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

This PR introduces a new GitHub workflow to check for manual changes to the `aws-amplify` package.json file. If changes are detected, the workflow will fail, add a comment on the PR, and block the PR until a repo admin can review & merge.

After this change is merged, the new action will be marked as `required` for the `main` branch.

#### Description of how you validated changes
Tested workflow on test repository: https://github.com/jimblanc/github-action-test

Example comment when the workflow fails:
Screenshot 2023-05-26 at 2 17 07 PM

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by cshfang — DISMISSED** [review]

**Review by AllanZhengYP — APPROVED** [review]

File: .github/workflows/aws-amplify-dependency-check.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, reopened]
+jobs:
+  checkUmbrellaDependencies:
+    permissions:
+      pull-requests: write # Used to add comment to PR
+    runs-on: ubuntu-latest
+    if: github.event.action == 'opened' || github.event.action == 'reopened'
```
Comment (elorzafe) [inline .github/workflows/aws-amplify-dependency-check.yml:11]: Is that redundant with types? or something required by github actions

**Review by elorzafe — COMMENTED** [review]
Why we cannot have `package.json` file changes to be approved by admin codeowners by default?

File: .github/workflows/aws-amplify-dependency-check.yml (hunk 1 above)
Comment (jimblanc) [reply .github/workflows/aws-amplify-dependency-check.yml]: Ah yes you're right, confirmed on my test repo. Made the change.

**Review by cshfang — APPROVED** [review]

File: .github/workflows/aws-amplify-dependency-check.yml (hunk 1 above)
Comment (jimblanc) [reply .github/workflows/aws-amplify-dependency-check.yml]: Regarding `CODEOWNERS`, we made that change as well, but this action was in response to specific feedback on the COE to make the reasons more clear to other engineers working with the repo (via the comment).

**Review by elorzafe — APPROVED** [review]

**Pull request merged**